src/structure.js

```diff
@@ -45,6 +46,7 @@ function analyzeType(value) {
           }
         }
         return $(new Type())
+          .map(value => isSugar(value) ? desugar(value) : value)
```

Comment:
Again, we can hide the `isSugar` behind the `desugar` function. That let's us map cleanly:

```js
return $(new Type())
  .map(desugar)
```

Reply:
done